src/RawPOSIX/src/safeposix/syscalls/sys_calls.rs

```diff
@@ -202,6 +202,11 @@ impl Cage {
      *   zombie list and retrieve the first entry from it (first in, first out).
      */
     pub fn waitpid_syscall(&self, cageid: i32, status: &mut i32, options: i32) -> i32 {
+        // 0) Check for pending signals and interrupt if any
```

Comment:
I think its probably superfluous to check at the beginning here since we'll quickly get to one of the checks in either of the wait states.